Localized fromFormatExplain with numberingSystem support

shripadk opened the pull request:
**Problem**: Running `fromFormatExplain` on a localized string with a numberingSystem defined would have `rawMatches` return `null` and `matches` an empty object:

<img width="847" alt="Screenshot 2019-04-03 at 1 15 25 PM" src="https://user-images.githubusercontent.com/189344/55461661-9de5e480-5612-11e9-9647-89ebc74d1aa0.png">

With this PR, it is now possible to parse localized numberingSystem strings correctly:

<img width="847" alt="Screenshot 2019-04-03 at 1 20 04 PM" src="https://user-images.githubusercontent.com/189344/55461907-4005cc80-5613-11e9-8cb6-ec4bcce6ea1c.png">

icambron commented on src/impl/tokenParser.js:
```diff
@@ -64,32 +69,32 @@ function unitForToken(token, loc) {
         case "MM":
           return intUnit(two);
         case "MMM":
-          return oneOf(loc.months("short", false, false), 1);
```
I don't see why these should change from false to true?

shripadk replied:
It is to bring parity with format string in `toFormat`. For instance, let us consider the locale to be `zh` and numberingSystem to be `hanidec`. It would be something like this with the above fix:

<img width="1304" alt="Screenshot 2019-04-06 at 10 31 23 AM" src="https://user-images.githubusercontent.com/189344/55665114-408ea500-5857-11e9-8bc1-3fb8cdd1f01c.png">

Earlier, `loc.months` returned an array of months in a different format which made it difficult to easily transpose the output of `toFormat` with `fromFormat`. The ideal situation would be that the user can feed the output of `toFormat` to `fromFormat` and get back the original value. I know that it doesn't work in some cases (like if the format has `EEEEE` making it ambiguous. does "T" stand for "Tuesday" or "Thursday"?).

From my tests, the output of:

`loc.months("long", false, false)` for `hanidec` numberingSystem was:

`[ '一月', '二月', '三月', '四月', '五月', '六月', '七月', '八月', '九月', '十月', '十一月', '十二月' ]`

As you can clearly see the `toFormat` just output the numerical equivalent of "April" as `四` but `loc.months` output the string equivalent of "April" as `四月`. The parser was unable to parse the string (because there is no `四月` in the localized string) causing it to output `null` for rawMatches. The correct output for `loc.months` should be:

`[ '一', '二', '三', '四', '五', '六', '七', '八', '九', '一〇', '一一', '一二' ]`

I think your intent was right but the flags were probably flipped in the implementation of `Locale.months` in locale.js. I was not sure about it so I did not touch the locale.js code. I instead flipped the boolean in the parser logic itself. It did not cause any other tests to fail so I am assuming it wasn't tested earlier and my tests were the first to check that line of code.

Nevertheless please do let me know if my understanding was incorrect. I am new to your library (which is absolutely amazing by the way) and am learning the internals on the go. 😃

icambron commented:
This totally makes sense and I'm not really sure how to improve the code from what you have. I'm just a little bummed that it's so much code. Let me think on this one a bit

shripadk commented:
@icambron Absolutely no issues! Please take your time and do let me know if you encounter something confusing in the code. Thanks!

shripadk commented:
Refactored `digits.js` file to only have a single source of truth for unicode endpoints for various numberingSystems. This should make maintenance easier if and when new numberingSystems are added by browser vendors.

Also added a test to ensure that future additions or changes always fall within the desired range (must map one-to-one with latin digits 0-9).

icambron commented:
I've decided I'm going to merge this

icambron commented:
Merged in cbc9e6e621241634fb03ee417af7a97000dd400f
